scripts/ia_healthcheck.py

```diff
@@ -0,0 +1,92 @@
+from datetime import datetime, timedelta
+import random
+from ..web_monitoring import db
+from ..web_monitoring import internetarchive
+
+
+MAX_CAPTURE_AGE = timedelta(hours=72)
+LINKS_TO_CHECK = 5
+
+
+def query_webmondb():
+    """
+    Query the web monitoring SCANNER API to get random links from the database
+    Interger -> List
+    """
+    try:
+        response = db.Client.from_env()
+    except Exception as e:
+        print('Could not access Web Monitoring API')
+        print(e)
+        return []
+    else:
+        page = response.list_pages(chunk=1, chunk_size=1)
+        url_count = page['meta']['total_results']
+        return ([response.list_pages(chunk=number, chunk_size=1)
+            ['data'][0]['url']
+            for number in random.sample(range(url_count), LINKS_TO_CHECK)])
+
+
+def query_waybackCDX(url):
+    """
+    Query Wayback Machine using CDX API
+    Get a list of versions of a url, and check for ValueError.
+    String -> JSON
+    """
+    try:
+        verisons = internetarchive.list_versions (url, from_date=datetime.now()
+            - MAX_CAPTURE_AGE)
+        verision = next(verisons)
+    except ValueError as e:
+        print(e)
+        status = False
+        return status
+    except:
+        print("Could not access Wayback CDX API")
+    else:
+        status = True
+        return status
+
+
+def output_file(responses):
+    """
+    Write Output to a Text file
+    List -> None
+    """
+    fileoutput = open("ia_healthcheck.txt", "w")
```

Comment:
You should either use a context manager:

```py
with open('ia_healthcheck.txt', 'w') as fileoutput:
    # code here
```

Or make sure any possible errors here are handled so you don’t leave the file handle open.

---

Also, maybe useful to make this path a CLI arg or at least an argument to this function so it can be configured.

OR! Skip worrying about a file altogether here and write to stdout; then someone calling this program can just `ia_healthcheck > somefile.txt` or `ia_healthcheck | grep 'False:'` or whatever. You’ll have to make sure all your other `print()` calls go to `stderr`, though.

Finally, maybe consider a more machine-readable format for the output here, e.g. YAML, or JSON? Dunno, this is simple enough it might be fine, too.

Reply:
Went with the 2nd suggestion, and writes to stdout.